src/main/java/bio/terra/service/search/SearchService.java

```diff
@@ -57,16 +62,30 @@ public SearchService(BigQueryPdao bigQueryPdao, RestHighLevelClient client) {
         this.client = client;
     }
 
+    private String uuidToIndexName(UUID id) {
+        return String.format("idx-%s", id);
+    }
+
+    private String translateToTim(String sql) {
+        Pattern regex = Pattern.compile("( as )(\\w+)");
+        Matcher matches = regex.matcher(sql.toLowerCase());
+        StringBuilder sb = new StringBuilder(sql.length());
+        while(matches.find()) {
+            String replacement = timMap.get(matches.group(2));
+            if (replacement != null) {
+                matches.appendReplacement(sb, matches.group(1) + TimUtils.encode(replacement));
```

Comment:
I'm not familiar with `appendReplacement()`, does it do the right thing is you don't replace every match you find? That is, does it keep track of the previously included match and know how to insert all the text leading up to the current one? I suppose if `TimUtils.encode()` can be applied in all cases then you'd be calling it on every form anyway.

Reply:
It does seem to do the right thing. I tried with some added `AS` columns and they weren't replaced.